Scripts/Runtime/Entities/Tasks/Data/JobTaskWorkConfig.cs

```diff
@@ -40,12 +41,20 @@ internal JobTaskWorkConfig(ScheduleJobDelegate scheduleJobDelegate, AbstractTask
         /// <typeparam name="TKey">The type of the key</typeparam>
         /// <typeparam name="TInstance">The type of the data</typeparam>
         /// <returns>This <see cref="JobTaskWorkConfig"/> for chaining additional configuration.</returns>
-        public JobTaskWorkConfig ScheduleOn<TKey, TInstance>(VirtualData<TKey, TInstance> data, BatchStrategy batchStrategy)
-            where TKey : unmanaged, IEquatable<TKey>
-            where TInstance : unmanaged, IKeyedData<TKey>
+        public JobTaskWorkConfig ScheduleOn<TInstance>(VirtualData<TInstance> data, BatchStrategy batchStrategy)
```

Comment:
```
ConfigureCancelJob(HandleCancelJob.Schedule)
               .ScheduleOnForCancel(System.WanderInstanceData, BatchStrategy.MaximizeChunk)
               .RequireCancelledDataForIterateAsync(System.WanderInstanceData)
               .RequireDataForAddAsync(RestartData);
```

Should we optimize this and automatically do the `ScheduleOnForCancel` part and `RequireCancelledDataForIterateAsync` for you?

```
ConfigureCancelJob(HandleCancelJob.Schedule, System.WanderInstanceData, BatchStrategy.MaximizeChunk)
               .RequireDataForAddAsync(RestartData);
```

It's a bit nicer/easier and you won't forget, but you don't have as many hints for writing your job about what you need to get to use.

Reply:
I'm going to need you to refresh me on what other types of `Configure*Job`s there are.
I like the first option better because it's more consistent. However having a unique `.ScheduleOnForCancel` method seems wrong. Why can't we use `ScheduleOn` for this?